### DISCUSSION TITLE
Device flow authentication

### DISCUSSION THREAD

**Discussion by kcranston** [community/community#24352] [Apps, API and Webhooks]
<p>I am following the instructions for <a href="https://docs.github.com/en/free-pro-team@latest/developers/apps/authorizing-oauth-apps#device-flow" rel="noopener nofollow ugc">Device Flow Authorization</a> and I can’t get Step 3 to work (apps polls for authentication). After successfully entering the user_code at the login/device URL, I then POST to<br>
the login/oauth/access_token and get this response:</p>
<pre><code class="lang-auto">{'error': 'incorrect_client_credentials', 'error_description': 'The client_id and/or client_secret passed are incorrect.', 'error_uri': 'https://docs.github.com/apps/managing-oauth-apps/troubleshooting-oauth-app-access-token-request-errors/#incorrect-client-credentials'}
</code></pre>
<p>Here is a gist of the python implementation that’s failing - the same client_id is being passed to both calls, and the docs don’t indicate that I need a client_secret.<br>
</p><aside class="onebox githubgist">
  <header class="source">
      <a href="https://gist.github.com/kcranston/6edfebf177f82af6c6f6e1db92e9c722" target="_blank" rel="noopener nofollow ugc">gist.github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://gist.github.com/kcranston/6edfebf177f82af6c6f6e1db92e9c722" target="_blank" rel="noopener nofollow ugc">https://gist.github.com/kcranston/6edfebf177f82af6c6f6e1db92e9c722</a></h4>
<h5>device-flow.py</h5>
<pre><code class="Python">import requests

client_id = "Iv1.8df72ad9560c774c"
header = {"Content-Type": "application/json", "Accept": "application/json"}

# Step 1
payload1 = {"client_id": client_id,}
r = requests.post(
    'https://github.com/login/device/code',headers=header,json=payload1
    )</code></pre>
This file has been truncated. <a href="https://gist.github.com/kcranston/6edfebf177f82af6c6f6e1db92e9c722" target="_blank" rel="noopener nofollow ugc">show original</a>

<p>
</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>This is for a GitHub App, created by me and installed on my account. I feel like I am missing something basic here.</p>

**Comment by karuppiah7890** [comment]
<p>I tried the same in curl in my local and it all worked</p>
<pre><code class="lang-bash">curl -i -X POST \
-H 'Content-Type: application/json' \
-d '{
    "client_id": "Iv1.8df72ad9560c774c"
}' \
https://github.com/login/device/code
</code></pre>
<pre><code class="lang-bash">curl -i -X POST \
-H 'Content-Type: application/json' \
-d '{
    "client_id": "Iv1.8df72ad9560c774c",
    "device_code": "afaccc750f9aeb3b7ab9ea592df97e06bffa6977",
    "grant_type": "urn:ietf:params:oauth:grant-type:device_code"
}' \
https://github.com/login/oauth/access_token
</code></pre>
<p>Maybe the code has some sort of issue</p>

**Comment by kcranston** [comment, marked as answer]
<p>You are right - this is working from the command line and was a problem with the code. Stupid syntax mistake - in the payload for Step 3, I had the key <code>"grant_type:"</code>  with an extra colon inside the quotes.  Sigh. Gist now updated.</p>
